src/pip/_internal/cli/progress_bars.py

```diff
@@ -145,24 +147,38 @@ def __init__(self, *args, **kwargs):
     @property
     def downloaded(self):
         # type: () -> str
-        return format_size(self.index)  # type: ignore
+
+        self.index = getattr(self, 'index')
+        return format_size(self.index)
 
     @property
     def download_speed(self):
         # type: () -> str
+
+        self.avg = getattr(self, 'avg')
```

Comment:
Since we don't have instance attributes like `index`, `avg` etc defined,  if we just remove the `# type: ignore` comments, mypy raises e.g. `error: "DownloadProgressMixin" has no attribute "avg"`

So either I explicity define these in the constructor, like `self.avg = <something>`, but then I don't know what initial value I assign them to, or do I just assign `None`, or I can use `getattr` to get the values from them when the property is accessed, and mypy stops complaining after that.

I think it's more of a design choice, since I saw `getattr` being used at other places as well

Reply:
Is there a tracking issue on mypy for this? If yes, I'd prefer that we keep the `# type: ignore` here, and add a link to the underlying mypy issue.

Reply:
Hi @pradyunsg ,

The closest  issue I could find was https://github.com/python/mypy/issues/5868, but I don't think this is related. 

I think it's a valid error thrown by mypy, since we haven't defined the instance attribute in the class, or used any base classes where it might have been defined, so mypy complains, so instead of explicitly defining them in the class, I used `getattr` to get the values for these attributes, and mypy doesn't complain after it

Reply:
Pinging @pradyunsg :)